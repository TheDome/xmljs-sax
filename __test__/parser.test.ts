import parse from "../lib";

describe("XMLParser", function () {
  it("should read a simple XML", async function () {
    const xml =
      '<?xml version="1.0"?>\n' +
      "<soap:Envelope\n" +
      'xmlns:soap="http://www.w3.org/2001/12/soap-envelope"\n' +
      'soap:encodingStyle="http://www.w3.org/2001/12/soap-encoding">\n' +
      "\n" +
      '  <soap:Body xmlns:m="http://www.example.org/stock">\n' +
      "    <m:GetStockPriceResponse>\n" +
      '      <m:Price includedVat="1.21">34.5</m:Price>\n' +
      "    </m:GetStockPriceResponse>\n" +
      "  </soap:Body>\n" +
      "\n" +
      "</soap:Envelope>";
    await expect(parse(xml)).resolves.toMatchSnapshot();
  });

  it("should read a complicated XML", async function () {
    const xml =
      '<?xml version="1.0"?>\n' +
      "<soap:Envelope\n" +
      'xmlns:soap="http://www.w3.org/2001/12/soap-envelope"\n' +
      'soap:encodingStyle="http://www.w3.org/2001/12/soap-encoding">\n' +
      "\n" +
      '  <soap:Body xmlns:m="http://www.example.org/stock">\n' +
      "    <m:GetStockPriceResponse>\n" +
      "      <m:Price>34.5</m:Price>\n" +
      "    </m:GetStockPriceResponse>\n" +
      "    <m:GetStockPriceResponse>\n" +
      "        <m:Price>35.5</m:Price>\n" +
      "        <m:priCe>36.5</m:priCe>\n" +
      "    </m:GetStockPriceResponse>\n" +
      "  </soap:Body>\n" +
      "\n" +
      "</soap:Envelope>";
    await expect(parse(xml)).resolves.toMatchSnapshot();
  });
});

describe("Resolver", function () {
  it("should resolve the path to a child", async function () {
    const xml =
      '<?xml version="1.0"?>\n' +
      "<soap:Envelope\n" +
      'xmlns:soap="http://www.w3.org/2001/12/soap-envelope"\n' +
      'soap:encodingStyle="http://www.w3.org/2001/12/soap-encoding">\n' +
      "\n" +
      '  <soap:Body xmlns:m="http://www.example.org/stock">\n' +
      "    <m:GetStockPriceResponse>\n" +
      '      <m:Price includedVat="1.21">34.5</m:Price>\n' +
      "    </m:GetStockPriceResponse>\n" +
      "  </soap:Body>\n" +
      "\n" +
      "</soap:Envelope>";

    const root = await parse(xml);

    const priceNode = root.resolveNSPath(
      "A:Envelope/A:Body/B:GetStockPriceResponse/B:Price",
      {
        A: "http://www.w3.org/2001/12/soap-envelope",
        B: "http://www.example.org/stock",
      }
    );

    expect(priceNode).not.toBeNull();
    expect(priceNode?.value).toEqual("34.5");
  });
});
