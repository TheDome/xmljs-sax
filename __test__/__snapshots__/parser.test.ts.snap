// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XMLParser should read a complicated XML 1`] = `
XmlNode {
  "_parent": null,
  "attributes": Array [
    Object {
      "local": "encodingStyle",
      "name": "soap:encodingStyle",
      "prefix": "soap",
      "uri": "http://www.w3.org/2001/12/soap-envelope",
      "value": "http://www.w3.org/2001/12/soap-encoding",
    },
  ],
  "childs": Array [
    XmlNode {
      "_parent": [Circular],
      "attributes": Array [],
      "childs": Array [
        XmlNode {
          "_parent": [Circular],
          "attributes": Array [],
          "childs": Array [
            XmlNode {
              "_parent": [Circular],
              "attributes": Array [],
              "childs": Array [
                XmlNode {
                  "_parent": [Circular],
                  "attributes": Array [],
                  "childs": Array [
                    XmlNode {
                      "_parent": [Circular],
                      "attributes": Array [],
                      "childs": Array [
                        XmlNode {
                          "_parent": [Circular],
                          "attributes": Array [],
                          "childs": Array [],
                          "fullName": "m:priCe",
                          "name": "priCe",
                          "namespace": Object {
                            "m": "http://www.example.org/stock",
                            "soap": "http://www.w3.org/2001/12/soap-envelope",
                          },
                          "value": "36.5",
                        },
                      ],
                      "fullName": "m:Price",
                      "name": "Price",
                      "namespace": Object {
                        "m": "http://www.example.org/stock",
                        "soap": "http://www.w3.org/2001/12/soap-envelope",
                      },
                      "value": "35.5",
                    },
                  ],
                  "fullName": "m:GetStockPriceResponse",
                  "name": "GetStockPriceResponse",
                  "namespace": Object {
                    "m": "http://www.example.org/stock",
                    "soap": "http://www.w3.org/2001/12/soap-envelope",
                  },
                },
              ],
              "fullName": "m:Price",
              "name": "Price",
              "namespace": Object {
                "m": "http://www.example.org/stock",
                "soap": "http://www.w3.org/2001/12/soap-envelope",
              },
              "value": "34.5",
            },
          ],
          "fullName": "m:GetStockPriceResponse",
          "name": "GetStockPriceResponse",
          "namespace": Object {
            "m": "http://www.example.org/stock",
            "soap": "http://www.w3.org/2001/12/soap-envelope",
          },
        },
      ],
      "fullName": "soap:Body",
      "name": "Body",
      "namespace": Object {
        "m": "http://www.example.org/stock",
        "soap": "http://www.w3.org/2001/12/soap-envelope",
      },
    },
  ],
  "fullName": "soap:Envelope",
  "name": "Envelope",
  "namespace": Object {
    "soap": "http://www.w3.org/2001/12/soap-envelope",
  },
}
`;

exports[`XMLParser should read a simple XML 1`] = `
XmlNode {
  "_parent": null,
  "attributes": Array [
    Object {
      "local": "encodingStyle",
      "name": "soap:encodingStyle",
      "prefix": "soap",
      "uri": "http://www.w3.org/2001/12/soap-envelope",
      "value": "http://www.w3.org/2001/12/soap-encoding",
    },
  ],
  "childs": Array [
    XmlNode {
      "_parent": [Circular],
      "attributes": Array [],
      "childs": Array [
        XmlNode {
          "_parent": [Circular],
          "attributes": Array [],
          "childs": Array [
            XmlNode {
              "_parent": [Circular],
              "attributes": Array [
                Object {
                  "local": "includedVat",
                  "name": "includedVat",
                  "prefix": "",
                  "uri": "",
                  "value": "1.21",
                },
              ],
              "childs": Array [],
              "fullName": "m:Price",
              "name": "Price",
              "namespace": Object {
                "m": "http://www.example.org/stock",
                "soap": "http://www.w3.org/2001/12/soap-envelope",
              },
              "value": "34.5",
            },
          ],
          "fullName": "m:GetStockPriceResponse",
          "name": "GetStockPriceResponse",
          "namespace": Object {
            "m": "http://www.example.org/stock",
            "soap": "http://www.w3.org/2001/12/soap-envelope",
          },
        },
      ],
      "fullName": "soap:Body",
      "name": "Body",
      "namespace": Object {
        "m": "http://www.example.org/stock",
        "soap": "http://www.w3.org/2001/12/soap-envelope",
      },
    },
  ],
  "fullName": "soap:Envelope",
  "name": "Envelope",
  "namespace": Object {
    "soap": "http://www.w3.org/2001/12/soap-envelope",
  },
}
`;
